### ISSUE TITLE
`FxGraphDrawer` fails on `einsum` nodes

### ISSUE THREAD

**Issue by f-dangel** [pytorch/pytorch#147884]
### 🐛 Describe the bug

I am trying to visualize a `torch.fx.GraphModule` using `torch.fx.passes.graph_drawer.FxGraphDrawer`.
If the graph module contains an `einsum` operation, I get a `bad label` error.

Here is an MWE to reproduce the problem:
```python
"""Cannot visualize `einsum` nodes with `torch.fx` graph drawer."""

from torch import einsum
from torch.fx import passes, symbolic_trace
from torch.nn import Module

# Setting this to `True` triggers the error.
# Everything works fine if set to `False`.
USE_EINSUM = True


class Square(Module):
    def forward(self, x):
        if USE_EINSUM:
            return einsum("a,a->a", x)
        else:
            return x**2


mod = symbolic_trace(Square())
g = passes.graph_drawer.FxGraphDrawer(mod, "einsum")
g.get_dot_graph().write_svg("einsum.svg")
``` 

Here is an example of the error:
```bash
b'Error: bad label format {name=%einsum|op_code=call_function\\n|target=torch.functional.einsum\\n|args=(a,a->a,)|num_users=1\\n}\n'
```

### Versions

```
PyTorch version: 2.4.0.post101
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 15.3.1 (x86_64)
GCC version: Could not collect
Clang version: 16.0.0 (clang-1600.0.26.6)
CMake version: version 3.31.0
Libc version: N/A

Python version: 3.9.16 (main, May 15 2023, 18:51:40)  [Clang 14.0.6 ] (64-bit runtime)
Python platform: macOS-10.16-x86_64-i386-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Apple M2

Versions of relevant libraries:
[pip3] flake8==7.1.1
[pip3] flake8-bugbear==24.12.12
[pip3] flake8-comprehensions==3.16.0
[pip3] flake8-tidy-imports==4.11.0
[pip3] mypy-extensions==1.0.0
[pip3] numpy==2.0.1
[pip3] torch==2.4.0.post101
[conda] libtorch                  2.4.0           cpu_mkl_hdbae018_101    conda-forge
[conda] mkl                       2023.2.0         h54c2260_50500    conda-forge
[conda] numpy                     2.0.2                    pypi_0    pypi
[conda] numpy-base                2.0.1            py39h03d8c7d_1  
[conda] pytorch                   2.4.0           cpu_mkl_py39hd2dbf71_101    conda-forge
```

**Comment by Chinmayrawat15** [comment]
Still reproduces on the current nightly (2.14.0.dev20260811) with graphviz 15.1.1:

```
Error: bad label format {name=%einsum|op_code=call_function\n|target=torch.functional.einsum\n|args=(a,a->a,)|num_users=1\n}
```

The cause is that `FxGraphDrawer` draws each node as a graphviz `record`, and record labels are structured by `{`, `}`, `|`, `<` and `>`. `_typename` and the args/kwargs formatting only escape `{` and `}`, so the `>` in the einsum equation ends the label early. Worth noting for anyone hitting this: dot still writes an SVG, the einsum node just silently loses all of its fields, so it can look like a rendering quirk rather than an error.

While fixing it I found two more label fields with the same problem that escape nothing at all, and break on the same class of input:

* a leaf module's `__constants__`, where a dict value renders as `{'a': 1}`
* the parsed stack trace, where both the source line (e.g. `if x.sum() > 0`) and pseudo-filenames like `<stdin>` are interpolated verbatim

I have a fix covering all three plus tests, and I'll open a PR shortly. Happy to take this if nobody else is on it.

I used Claude Code to help investigate and write the change; the error output above is verbatim program output, and I've reviewed and tested the fix myself.
